src/qgis_geonode/qgisgeonode/conf.py

```diff
@@ -0,0 +1,85 @@
+import contextlib
+import logging
+import typing
+
+from qgis.PyQt import QtCore
+from qgis.core import QgsSettings
+
+logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def qgis_settings(group_root: str):
+    """A simple context manager to help managing our own settings in QgsSettings"""
+    settings = QgsSettings()
+    settings.beginGroup(group_root)
+    try:
+        yield settings
+    finally:
+        settings.endGroup()
+
+
+class SettingsManager(QtCore.QObject):
+    """Manage saving/loading settings for the plugin in QgsSettings"""
+
+    BASE_GROUP_NAME: str = "qgis_geonode"
+    SELECTED_CONNECTION_KEY: str = "selected_connection"
+
+    current_connection_changed = QtCore.pyqtSignal(str)
+
+    def list_connections(self) -> typing.List[str]:
+        with qgis_settings(f"{self.BASE_GROUP_NAME}/connections") as settings:
+            result = settings.childGroups()
+        result.sort()
+        return result
+
+    def get_connection_settings(self, name) -> typing.Dict:
+        details = {}
+        with qgis_settings(self._get_connection_settings_base(name)) as settings:
+            for key in settings.allKeys():
+                details[key] = settings.value(key)
+        if len(details) == 0:
+            raise ValueError(
+                f"Could not find a connection named {name!r} in QgsSettings"
+            )
+        details["name"] = name
+        return details
+
+    def save_connection_settings(self, name, **additional_settings):
+        with qgis_settings(self._get_connection_settings_base(name)) as settings:
+            for name, value in additional_settings.items():
+                settings.setValue(name, value)
```

Comment:
when editing a connection, we might need to delete the past connection settings here. I have tested this on QGIS, another connection is created when editing a connection name.

Reply:
I solved this by adding an additional `id` to each `ConnectionSettings` instance. This `id` field is not shown to the user and remains stable